src/library/scala/collection/immutable/Vector.scala

```diff
@@ -1200,6 +1217,17 @@ private[immutable] trait VectorPointer[+T] {
     }
 
 
+  /**
+    *
+    * @param array
+    * @return
+    */
+    @`inline` private def safeClone(array: Array[AnyRef]): Array[AnyRef] = {
```

Comment:
`Arrays.copyOf` is simpler and probably faster (unless the JIT can optimize the bounds checks and nulling away for your implementation)

Reply:
See https://github.com/scala/scala/pull/8276#discussion_r308016475